### PR TITLE
bal 2495 - ballerine new api plugins

### PR DESCRIPTION
**Pull request by liorzam**
## Summary by CodeRabbit

- **New Features**
  - Introduced an API plugin for Ballerine service, enhancing workflow management capabilities.
  - Added endpoints to manage workflow plugins, including listing and adding plugins.
  
- **Improvements**
  - Enhanced error handling and validation for API responses, specifically for unauthorized access.
  - Improved control flow and error handling in various service methods, ensuring more robust operations.
  
- **Documentation**
  - Updated Swagger API documentation to reflect new endpoints and response schemas.

- **Chores**
  - Refactored and streamlined type definitions and method structures for improved maintainability.

### CODE REVIEW

File: packages/workflow-core/src/lib/plugins/common-plugin/vendor-consts.ts [hunk 1]
```diff
@@ -6,14 +6,81 @@ export const SANCTIONS_SCREENING_VENDOR = {
...
+  CompanySanctionsOptions &
+  UboOptions &
+  KybOptions;
+
 export const SANCSIONS_SCREENING_BASE_RESPONSE = {
   transform: [
     {
       mapping:
-        "merge({ name: 'company_sanctions', status: reason == 'NOT_IMPLEMENTED' && 'CANCELED' || error != `null` && 'ERROR' || 'SUCCESS' }, @)",
+        "merge({ name: 'sanctions_screening', status: reason == 'NOT_IMPLEMENTED' && 'CANCELED' || error != `null` && 'ERROR' || 'IN_PROGRESS' }, @)",
```
Comment (alonp99) [inline packages/workflow-core/src/lib/plugins/common-plugin/vendor-consts.ts]: Lets talk about this change online

File: packages/workflow-core/src/lib/plugins/common-plugin/vendor-consts.ts [hunk 2]
```diff
@@ -6,14 +6,81 @@ export const SANCTIONS_SCREENING_VENDOR = {
...
+  request: SerializableValidatableTransformer;
+  response: SerializableValidatableTransformer;
+};
+
+type ApiPluginOptionsBase = {
+  pluginKind: ApiBallerinePlugins;
+};
+
+type DowJonesOptions = {
+  pluginKind: 'dow-jones';
```
Comment (alonp99) [inline packages/workflow-core/src/lib/plugins/common-plugin/vendor-consts.ts]: this should be "vendor" not plugin kind

plugin kind will be for example: "company_sanctions", "individual_sanctions", "kyc", "registry_information" etc..

File: packages/workflow-core/src/lib/plugins/common-plugin/vendor-consts.ts (hunk 1 above)
Comment (liorzam) [reply packages/workflow-core/src/lib/plugins/common-plugin/vendor-consts.ts]: it was mistake its for sanctions screening

File: packages/workflow-core/src/lib/plugins/common-plugin/vendor-consts.ts (hunk 2 above)
Comment (liorzam) [reply packages/workflow-core/src/lib/plugins/common-plugin/vendor-consts.ts]: Ok, will change it

**Pull request merged**